### ISSUE TITLE
Unable to authenticate with GitHub provider

### ISSUE THREAD

**Issue by MikeTheCanuck** [bvaughn/personal-logger#18]
GitHub authentication is failing when running the app in development mode (i.e. launched locally via `yarn start`) and in production (i.e. when attempting to authenticate at https://personal-logger.now.sh).

This has me blocked from being able to test my work on #17.

## Repro Steps
0. Use OS 10.13.2, Chrome 63; GitHub account with 2FA enabled.
1. Fork the `personal-logger` repo.
2. (I'd run `yarn install` a while back, then `git remote.../git fetch.../git merge...` after subsequent merges to master).
3. Run `git checkout -b mikethecanuck/disable-autocap` - but no changes to the code.
4. Run `yarn start`, browse to http://localhost:3000/ in Chrome
5. Select "Sign in with GitHub"
6. Authorize `bvaughn` for GitHub authN

## Expected Result
* app starts up, presenting the `header`, `nav` and `main` elements (good)

## Actual Result
* GitHub authorization for this app appears to succeed, then returns user to the "GitHub/Twitter" sign-in choices
* on all subsequent sign-in attempts, app redirects directly back to the GitHub/Twitter choice
* console briefly flashes the following two Uncaught TypeError exceptions and then clears (I had to video and pause to be able to transcribe the exception contents):
```
Uncaught TypeError: Cannot read property 'appendChild' of undefined
  at injectScript (content-script.js:10)
  at content-script.js:34
```
and
```
Uncaught TypeError: Cannot read property 'version' of undefined
  at installHook (installHook.js:29)
  at installHook.js:53
```

I don't know if there's some silent error that occurs when authorizing the `food-tracker-c6279.firebaseapp.com` authDomain.

I'll dig around those scripts but figured I'd report this first in case anyone else was hitting this hurdle.

**Comment by MikeTheCanuck** [comment]
Interesting - Chrome Dev Tools, "Sources/Network" tab had the `installHook.js` script open, and lines 29-53 look like this:
```
  if (instance.version) {
    __ReactSight_ReactVersion = instance.version;
    devTools.onCommitFiberRoot = (function (original) {
      return function (...args) {
        __ReactSightFiberDOM = args[1];
        traverse16();
        return original(...args);
      };
    })(devTools.onCommitFiberRoot);
  } else {
    // hijack receiveComponent method which runs after a component is rendered
    instance.Reconciler.receiveComponent = (function (original) {
      return function (...args) {
        if (!__ReactSightThrottle) {
          __ReactSightThrottle = true;
          setTimeout(() => {
            getData();
            __ReactSightThrottle = false;
          }, 10);
        }
        return original(...args);
      };
    })(instance.Reconciler.receiveComponent);
  }
})();
```

I have the [React-Sight](https://chrome.google.com/webstore/detail/react-sight/aalppolilappfakpmdfdkpppdnhpgifn) Chrome Extension installed and active.

When I disable the `React-Sight` extension, the app still fails to authenticate, but I no longer observe the console errors.  Instead, GitHub authentication just fails silently, redirecting back to the GitHub/Twitter choices page after a couple of seconds.

**Comment by MikeTheCanuck** [comment]
Q to @bvaughn: would you prefer a separate Issue be opened to address the silent GitHub authN failure, or would you prefer I rename this one according to the updated understanding of the core problem?

I'm cool going either way, depending on your approach to issue management.

**Comment by bvaughn** [comment]
I'm not sure what the "separate issue" would be, aside from the fact that you seem to be having trouble authenticating with GitHub while running this app locally.

Given that auth is handled by Firestore, and I've only ever used that lib for this project, I'm not sure how much insight I have to share at the moment. I will say that auth works fine for me on localhost in Chrome 62 on OS X and in Safari on iOS.

Is it possible you could use the Twitter auth method to unblock yourself for #17?

**Comment by MikeTheCanuck** [comment]
Fair enough - some projects prefer clean issue reports without irrelevant details polluting an issue, so I wanted to offer to accommodate.

I have only ever used the Google auth provider for my Firebase project so I'm even less sure what insight I can bring so far.  I'll update the original repro report with my own environment (similar to yours).

Twitter auth turns out to work in Chrome 63 and Safari on OS 10.13, so I'll proceed that way.

**Comment by MikeTheCanuck** [comment]
When attempting GitHub authorization on Safari on OS 10.13, the JavaScript console reports the following:

```
login:0
Refused to execute a script because its hash, its nonce, or 'unsafe-inline' does not appear in the script-src directive of the Content Security Policy.
```

Then returns to the "Sign in using..." selection screen that I started with (as I observe in Chrome 63).

**Comment by MikeTheCanuck** [comment]
Updated original report to reflect that - now that https://personal-logger.now.sh is using the GitHub/Twitter authentication options - I can now report that GitHub authentication is not working for me.

Oh heck, one thing just occurred to me - I have two-factor auth enabled on my GitHub account.  That's probably the culprit - many OAuth and similar hand-offs choke when 2FA is enabled.

**Comment by bvaughn** [comment]
Hm! I also have 2fa enabled on my GitHub account and it doesn't prevent
auth for me. Odd!

I had Google auth option enabled but it didn't persists between pages.
However now that I've updated the app to use the HashRouter, that wouldn't
be a big blocker. I could reenable it...

**Comment by bvaughn** [comment]
I'm not sure if this issue is worth keeping open (on this repo at least) since auth is handled by an external library (Firebase)- particularly if the issue is being caused by 2fa. Any thoughts on your end?

**Comment by bvaughn** [comment]
I'm closing this for now. 😄  We can talk more if there's concern on your end!
